### DISCUSSION TITLE
Replaced SSL-certificate -> accepted, but not used

### DISCUSSION THREAD

**Discussion by ejgh-oe** [Security-Onion-Solutions/securityonion#12246] [2.4]
### Version

2.4.40

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

32

### Storage for /

500

### Storage for /nsm

500

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi,
In order to get rid of the browser complaining about "untrusted SSL certificate", I tried to replace the default cert with a Cert from our internal CA following the guidelines under https://docs.securityonion.net/en/2.4/nginx.html#replacing-default-cert
Uploading both the .CRT and and the Key file went OK (no errors displayed). 
When I tried accessing the GUI after a couple of minutes while having closed and reopened my browser beforehand still the same default cert was used. I even tried "so-nginx-restart" as well as rebooting the manager node - didn't change a thing; still the default cert.

Any ideas as to why SO (i.e. nginx behind) accepted the custom cert, but doesn't use it?

Thanks much in advance for your help.
-ewald

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by petiepooo** [comment]
I'm not sure if Salt monitors for changes to the cert file.  Try running `sudo so-nginx-restart` at the bash prompt...

**Reply by ejgh-oe** [reply]
Hi,
As mentioned above in my original post, I already tried restarting nginx and even rebooted the manager node - still came up with the old (default) cert. 
Ruling out any browser issues I issued a `openssl s_client -connect manager-node:443` - it's definitely the default cert.

BTW, the config via the GUI shows the correct cert.

**Reply by petiepooo** [reply]
i missed that; sorry.  I'm out of ideas.

**Comment by cm-ops** [comment]
If you run `salt-call pillar.get nginx`, do you see your cert and key with replace default cert set to `true`?

**Reply by ejgh-oe** [reply]
Hi, 
Looks like it's set:

```
# salt-call pillar.get nginx
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
local:
    ----------
    enabled:
        True
    ssl:
        ----------
        replace_cert:
            True
#


```

**Reply by cm-ops** [reply]
Did that also show the correct cert and key?

**Reply by ejgh-oe** [reply]
Nope - that's all wrt. output that came from `salt-call pillar.get nginx`

**Reply by cm-ops** [reply]
In SOC, under nginx, what does the below show? Does it have your full cert/key in the settings?
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63416267/b4357487-038e-4883-a41d-e687ff341405)

**Reply by ejgh-oe** [reply]
Yes, absolutely:

For the cert
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/22fb02d9-c9f4-44d1-887a-9993b6a2a47c)
as well as for the key
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/a329c896-3269-408d-ad52-f5d13668c49a)

both ending with `-----END CERTIFICATE-----` and `-----END RSA PRIVATE KEY-----` respectively as last line

**Reply by cm-ops** [reply]
Do those match the `managerssl.crt` and `managerss.key` in `/etc/pki`?

**Reply by ejgh-oe** [reply]
Yes, absolutely - the files in /etc/pki are exactly the same contentwise as the ones I uploaded via the GUI.
```
# ls -l managerssl.*
-rw-r--r--. 1 root root   2142 Jan 24 09:17 managerssl.crt
-rw-r-----. 1 root socore 3246 Jan 24 09:17 managerssl.key
#
```

**Reply by cm-ops** [reply]
That all looks correct then, compare `/etc/pki/managerssl.*` against what is in the container - `docker exec -t so-nginx sh -c "ls -l /etc/pki/nginx/server/crt"` 

What does the cert show in the browser?

**Reply by ejgh-oe** [reply]
> That all looks correct then, compare /etc/pki/managerssl.* against what is in the container - `docker exec -t so-nginx sh -c "ls -l /etc/pki/nginx/server/crt"`



```
# docker exec -t so-nginx sh -c "ls -l /etc/pki/nginx/server/crt"
ls: /etc/pki/nginx/server/crt: No such file or directory
#
```
I changed tha above command to read `docker exec -t so-nginx sh -c "ls -l /etc/pki/nginx"
`
and got


```
# docker exec -t so-nginx sh -c "ls -l /etc/pki/nginx"
total 8
-rw-r--r--    1 root     root          2142 Jan 24 09:17 server.crt
-rw-r-----    1 root     939           3246 Jan 24 09:17 server.key
#
```

which are the same files as the ones under /etc/pki/managerssl.* (see my previous posting)

> What does the cert show in the browser?

It's the default-Cert that comes with SO

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/63328f5b-85fb-4e17-8b28-6e6657747c82)

("onion-mgr" is the name of the machine (manager-node)).

**Reply by cm-ops** [reply]
Have you tried to revert to default, run` so-checkin` and restart nginx, and add your custom `.crt/.key`, and run `so-checkin` then restart nginx container?

**Reply by ejgh-oe** [reply]
Just reverted everything to default in the GUI, i.e.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/4f04bf12-94eb-4956-9692-5d31246e7341)

and deleted the custom cert & key:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/6502a34a-ebcd-4344-87ba-58796b43ae97)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/ec1e2dcd-5fd3-4827-bd1e-b9ebc8c76569)

`so-checkin` ran for a while and ended with
```

...
Summary for local
--------------
Succeeded: 691 (changed=39)
Failed:      0
--------------
Total states run:     691
Total run time:    61.796 s
#
```

Connected to the GUI again - as expected default certificate.

Next I uploaded both, cert and key via the GUI again and set "Replace Default Cert" to true again. Re-ran `so-checkin` which ended with


```
...
Summary for local
--------------
Succeeded: 690 (changed=42)
Failed:      0
--------------
Total states run:     690
Total run time:    78.814 s
#
```


...but again, my custom certificate didn't get picked up:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/00c262fb-c8f2-4eb6-a103-9d9e7dbae9bb)

PS: Please note that this "custom certificate" is the same I had on my previous (2.3) installation of SO; also the machine name is the same.

**Comment by ejgh-oe** [comment]
Just to close this question: In the end I re-issued the Cert, included the new one in the SO-config and now it's working :-)
